cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -385,14 +389,24 @@ public void testDeployVersionedAndNonVersionedApp() throws Exception {
     Assert.assertEquals(versions, receivedVersions);
 
     appDetails = getAppDetails(appId.getNamespace(), appId.getApplication(), appIdVersion);
-    Assert.assertEquals(GSON.toJson(config), appDetails.getConfiguration());
     Assert.assertEquals(appIdVersion, appDetails.getAppVersion());
+    // when LCM is enabled a new app version is generated else -SNAPSHOT is overwritten
+    if (cConf.getBoolean(FEATURE_FLAG_PREFIX + Feature.LIFECYCLE_MANAGEMENT_EDIT.getFeatureFlagString())) {
```

Comment:
how come we're doing this manually instead of using the FeatureFlagsProvider?

Reply:
Also, why are we checking the feature flag, won't it always be the same value? It seems more appropriate to have two separate tests, one configured for LCM and one without.